tests/infer/test_inference.py

```diff
@@ -218,59 +218,51 @@ def guide():
-        self.do_elbo_test(fakes.NonreparameterizedGamma, 10000)
-
-    def do_elbo_test(self, gamma_dist, n_steps, trace_graph=False):
-        pyro.clear_param_store()
-
-        def model():
-            lambda_latent = pyro.sample("lambda_latent", gamma_dist(self.alpha0, self.beta0))
-            with pyro.iarange("data", self.n_data):
-                pyro.sample("obs", dist.Exponential(lambda_latent), obs=self.data)
-            return lambda_latent
-
-        def guide():
-            alpha_q_log = pyro.param(
-                "alpha_q_log",
-                torch.tensor(self.log_alpha_n.data + 0.17, requires_grad=True))
-            beta_q_log = pyro.param(
-                "beta_q_log",
-                torch.tensor(self.log_beta_n.data - 0.143, requires_grad=True))
-            alpha_q, beta_q = torch.exp(alpha_q_log), torch.exp(beta_q_log)
-            pyro.sample("lambda_latent", gamma_dist(alpha_q, beta_q))
-
-        adam = optim.Adam({"lr": .0003, "betas": (0.97, 0.999)})
-        svi = SVI(model, guide, adam, loss="ELBO", trace_graph=trace_graph)
-
-        for k in range(n_steps):
-            svi.step()
-
-        alpha_error = param_abs_error("alpha_q_log", self.log_alpha_n)
-        beta_error = param_abs_error("beta_q_log", self.log_beta_n)
-        assert_equal(0.0, alpha_error, prec=0.08)
-        assert_equal(0.0, beta_error, prec=0.08)
+@pytest.mark.parametrize('elbo_impl', ["Trace", "TraceGraph", "TraceEnum"])
+@pytest.mark.parametrize('gamma_dist,n_steps', [
+    (dist.Gamma, 5000),
+    (fakes.NonreparameterizedGamma, 10000),
+    (ShapeAugmentedGamma, 5000),
+], ids=['reparam', 'nonreparam', 'rsvi'])
+def test_exponential_gamma(gamma_dist, n_steps, elbo_impl):
+    pyro.clear_param_store()
+
+    # gamma prior hyperparameter
+    alpha0 = torch.tensor(1.0)
+    # gamma prior hyperparameter
+    beta0 = torch.tensor(1.0)
+    n_data = 2
+    data = torch.tensor([3.0, 2.0])  # two observations
+    alpha_n = alpha0 + torch.tensor(n_data)  # posterior alpha
+    beta_n = beta0 + torch.sum(data)  # posterior beta
+    log_alpha_n = torch.log(alpha_n)
+    log_beta_n = torch.log(beta_n)
+
+    def model():
+        lambda_latent = pyro.sample("lambda_latent", gamma_dist(alpha0, beta0))
+        with pyro.iarange("data", n_data):
+            pyro.sample("obs", dist.Exponential(lambda_latent), obs=data)
+        return lambda_latent
+
+    def guide():
+        alpha_q_log = pyro.param("alpha_q_log", torch.tensor(log_alpha_n.data + 0.17, requires_grad=True))
+        beta_q_log = pyro.param("beta_q_log", torch.tensor(log_beta_n.data - 0.143, requires_grad=True))
+        alpha_q, beta_q = torch.exp(alpha_q_log), torch.exp(beta_q_log)
+        pyro.sample("lambda_latent", gamma_dist(alpha_q, beta_q))
+
+    adam = optim.Adam({"lr": .0003, "betas": (0.97, 0.999)})
+    svi = SVI(model, guide, adam, loss="ELBO",
+              trace_graph=(elbo_impl == "TraceGraph"),
+              enum_discrete=(elbo_impl == "TraceEnum"),
```

Comment:
Note that there's nothing discrete here; we're simply testing that DiCE correctly handles RSVI's score_parts.

Reply:
is it not worth having a KL gradient test that mixes rsvi with discretes?

Reply:
Sure, I'll add some more test with rsvi + discrete enumeration

Reply:
Do you want this merged now or after you add the tests?